src/autoscaler/eventgenerator/generator/evaluator_test.go

```diff
@@ -467,10 +527,66 @@ var _ = Describe("Evaluator", func() {
 					})
 				})
 			})
+
+			Context("multiple triggers", func() {
+				BeforeEach(func() {
+					scalingEngine.RouteToHandler("POST", urlPath, ghttp.RespondWith(http.StatusOK, "successful"))
+					Expect(triggerChan).To(BeSent(triggerArrayMultipleTriggers))
+				})
+				Context("when only the first trigger breaches", func() {
+					BeforeEach(func() {
+						database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
+							return appMetricGEBreach, nil
+						}
+					})
+					It("should send trigger alarm to scaling engine", func() {
+						Eventually(scalingEngine.ReceivedRequests).Should(HaveLen(1))
+						Eventually(logger.LogMessages).Should(ContainElement(ContainSubstring("send trigger alarm to scaling engine")))
+					})
+				})
+
+				Context("when only second tigger breaches", func() {
+					BeforeEach(func() {
+						database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
+							return appMetricLEBreach, nil
+						}
+					})
+					It("should send trigger alarm to scaling engine", func() {
+						Eventually(scalingEngine.ReceivedRequests).Should(HaveLen(1))
+						Eventually(logger.LogMessages).Should(ContainElement(ContainSubstring("send trigger alarm to scaling engine")))
+					})
+				})
+
+				Context("when both tiggers breach", func() {
+					BeforeEach(func() {
+						database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
+							return appMetricMultipleTriggerAllBreach, nil
+						}
+					})
+					It("should send one trigger alarm to scaling engine", func() {
+						Eventually(scalingEngine.ReceivedRequests).Should(HaveLen(1))
```

Comment:
It depends on how we handle trigger conflict.  There are two options
1) first breach wins
2) trigger scaling for all the breaches
for 2, usually the second one will be ignored since there is cool down period control.  The scaling action can be taken only if the previous triggers did not trigger actual scaling for whatever reason(failed or ignored)